core/src/test/scala/integration/kafka/api/AbstractConsumerTest.scala

```diff
@@ -346,6 +346,7 @@ abstract class AbstractConsumerTest extends BaseRequestTest {
                                            partitionsToAssign: Set[TopicPartition],
                                            userRebalanceListener: ConsumerRebalanceListener)
     extends ShutdownableThread("daemon-consumer-assignment", false) {
+    setDaemon(true)
```

Comment:
Should we just add a parameter to ShutdownableThread to set the thread as daemon?

Reply:
It seems like one property of shutdownable thread is that we make it non-daemon and we need to explicitly shut it down. Were the try blocks not enough to fix the issue? Alternatively, can we make this a thread class that isn't shutdownable?